go/porcelain/deploy.go

```diff
@@ -626,6 +636,38 @@ func bundle(functionDir string, observer DeployObserver) (*deployFiles, error) {
 	return functions, nil
 }
 
+func bundleFromManifest(manifestFile *os.File, observer DeployObserver) (*deployFiles, error) {
+	manifestBytes, _ := ioutil.ReadAll(manifestFile)
+
+	var manifest functionsManifest
+
+	err := json.Unmarshal(manifestBytes, &manifest)
+
+	if err != nil {
+		return nil, fmt.Errorf("malformed functions manifest file: %w", err)
+	}
+
+	functions := newDeployFiles()
+
+	for _, function := range manifest.Functions {
+		fileInfo, err := os.Stat(function.Path)
+
+		if err != nil {
+			return nil, err
```

Comment:
Sounds good. Done in https://github.com/netlify/open-api/pull/362/commits/d5f62ae97270c7029688816872a8853230483060.

Would you still include `err` in the message? It felt a bit redundant to me, but I'm happy to add it if you think it can be useful.